File: Common/Models/PushNotification/PushNotificationMessageModel.cs
using System;
using IOBootstrap.NET.Common.Models.Base;

namespace IOBootstrap.NET.Common.Models.PushNotification;

public class PushNotificationMessageModel : IOModel
{
    public int ID { get; set; }
    public int DeviceType { get; set; }
    public string? NotificationCategory { get; set; }
    public string? NotificationData { get; set; }
    public string? NotificationMessage { get; set; }
    public string? NotificationTitle { get; set; }
    public bool IsCompleted { get; set; }
    public string? CreatedBy { get; set; }
    public DateTimeOffset CreatedDate { get; set; }
    public DateTimeOffset? UpdateDate { get; set; }
    public int? DeliveredDevicesCount { get; set; }
}

File: React.UI/modules/iobootstrap-bo-pushnotifications/src/models/PushNotificationMessageModel.ts
class PushNotificationMessageModel {

    id: number;
    deviceType: number | null;
    notificationCategory: string | null;
    notificationData: string | null;
    notificationMessage: string | null;
    notificationTitle: string;
    isCompleted: boolean;
    createdBy: string | null;
    createdDate: string | null;
    updateDate: string | null;
    deliveredDevicesCount: number | null;
    
    constructor() {
        this.id = 0;
        this.deviceType = null;
        this.notificationCategory = null;
        this.notificationData = null;
        this.notificationMessage = "";
        this.notificationTitle = "";
        this.isCompleted = false;
        this.createdBy = null;
        this.createdDate = null;
        this.updateDate = null;
        this.deliveredDevicesCount = null;
    }
}

export default PushNotificationMessageModel;
